modules/pool/lib/staking/gauge-staking.service.ts

```diff
@@ -180,40 +190,84 @@ export class GaugeStakingService implements PoolStakingService {
+        const childChainAbi = [
+            'function inflation_rate(uint256 week) view returns (uint256)',
+            'function working_supply() view returns (uint256)'
+        ];
+        const multicall = new Multicaller3(childChainAbi);
 
         gaugeAddresses.forEach((address) => {
-            // Only L2 gauges have the inflation_rate with a week parameter
-            if (networkContext.chain === 'MAINNET') {
-                multicall.call(address, address, 'inflation_rate', [], true);
-            } else {
-                multicall.call(address, address, 'inflation_rate', [currentWeek], true);
-            }
+            multicall.call(`${address}.inflationRate`, address, 'inflation_rate', [currentWeek], true);
+            multicall.call(`${address}.workingSupply`, address, 'working_supply', [], true);
         });
 
-        const childChainData = (await multicall.execute()) as Record<string, string | undefined>;
-
-        for (const childChainGauge in childChainData) {
-            if (childChainData[childChainGauge]) {
-                response[childChainGauge] = {
-                    version: 2,
-                    rate: formatUnits(childChainData[childChainGauge]!, 18),
-                };
-            } else {
-                response[childChainGauge] = {
-                    version: 1,
-                    rate: '0.0',
-                };
+        const results = (await multicall.execute()) as {
+            [address: string]: {
+                inflationRate: BigNumber | undefined,
+                workingSupply: BigNumber
             }
-        }
+        };
+
+        const response = Object.keys(results).reduce((acc, address) => {
+            const rate = results[address].inflationRate ? formatUnits(results[address].inflationRate!) : '0.0';
+            const workingSupply = formatUnits(results[address].workingSupply);
+            const version = results[address].inflationRate ? 2 : 1;
+            acc[address] = { version, rate, workingSupply };
+            return acc;
+        }, {} as { [gaugeAddress: string]: GaugeRate });
+
+        return response;
+    }
+
+    /**
+     * Get the inflation rate for all the gauges on mainnet.
+     * Gauges on mainnet use the inflation rate from token admin contract
+     * and relative weight from the gauge contract.
+     *
+     * @param gaugeAddresses 
+     * @returns 
+     */
+    private async getMainnetGaugeRates(gaugeAddresses: string[]): Promise<{ [gaugeAddress: string]: GaugeRate }> {
+        const now = Math.round(new Date().getTime() / 1000);
+        const version = 2; // On Mainnet BAL is always distributed directly to gauges
+        const { gaugeControllerAddress } = networkContext.data;
+        const abi = [
+            'function working_supply() view returns (uint256)',
+            'function gauge_relative_weight(address) view returns (uint256)'
+        ];
+        const multicall = new Multicaller3(abi);
```

Comment:
the gaugeController ABI is present under modules/vebal/abi . I'd suggest to move to modules/web3/abi and import from there. If we do this inline like this, I'm sure at some point the gaugeController changes and these functions are different

Reply:
changed to the ABI, we might need to do some work anyways when it changes.

Reply:
this import seems to be breaking in the github workflow, would you know why?